src/qs8-packw/gen/qs8-packw-x8c4-gemm-goi-avxvnni-prfm.c

```diff
@@ -0,0 +1,412 @@
+        v0 = _mm256_blend_epi32(v0, _mm256_set1_epi32((int32_t) unaligned_load_u32(w6)), 0x40);
+        v0 = _mm256_blend_epi32(v0, _mm256_set1_epi32((int32_t) unaligned_load_u32(w7)), 0x80);
+        xnn_prefetch_to_l1((const int8_t*) w0 + 448);
+        xnn_prefetch_to_l1((const int8_t*) w1 + 448);
+        xnn_prefetch_to_l1((const int8_t*) w2 + 448);
+        xnn_prefetch_to_l1((const int8_t*) w3 + 448);
+        xnn_prefetch_to_l1((const int8_t*) w4 + 448);
+        xnn_prefetch_to_l1((const int8_t*) w5 + 448);
+        xnn_prefetch_to_l1((const int8_t*) w6 + 448);
+        xnn_prefetch_to_l1((const int8_t*) w7 + 448);
+
+        vacc0 = _mm256_dpbusd_avx_epi32(vacc0, vone, v0);
+
+        _mm256_storeu_si256((__m256i *)&out[0],  v0);
+
+        w0 += 4;
+        w1 += 4;
+        w2 += 4;
+        w3 += 4;
+        w4 += 4;
+        w5 += 4;
+        w6 += 4;
+        w7 += 4;
+        out += 32;
+      }
+
+      // KC remainder of 1..3
+      if (k != 0) {
+        assert(k >= 1 && k <= 3);
+
+        const __m256i vmask = _mm256_set1_epi32((1u << (k * sizeof(int8_t) * 8)) - 1);
+
+        __m256i v0 = _mm256_set1_epi32((int32_t) safe_load_32bits(w0, k));
+        v0 = _mm256_blend_epi32(v0, _mm256_set1_epi32((int32_t) safe_load_32bits(w1, k)), 0x02);
+        v0 = _mm256_blend_epi32(v0, _mm256_set1_epi32((int32_t) safe_load_32bits(w2, k)), 0x04);
+        v0 = _mm256_blend_epi32(v0, _mm256_set1_epi32((int32_t) safe_load_32bits(w3, k)), 0x08);
+        v0 = _mm256_blend_epi32(v0, _mm256_set1_epi32((int32_t) safe_load_32bits(w4, k)), 0x10);
+        v0 = _mm256_blend_epi32(v0, _mm256_set1_epi32((int32_t) safe_load_32bits(w5, k)), 0x20);
+        v0 = _mm256_blend_epi32(v0, _mm256_set1_epi32((int32_t) safe_load_32bits(w6, k)), 0x40);
+        v0 = _mm256_blend_epi32(v0, _mm256_set1_epi32((int32_t) safe_load_32bits(w7, k)), 0x80);
+        v0 = _mm256_and_si256(v0, vmask);
+
+        w0 += k;
+        w1 += k;
+        w2 += k;
+        w3 += k;
+        w4 += k;
+        w5 += k;
+        w6 += k;
+        w7 += k;
+
+        vacc0 = _mm256_dpbusd_avx_epi32(vacc0, vone, v0);
+
+        _mm256_storeu_si256((__m256i *)&out[0],  v0);
+
+        out += 32;
+      }
+
+      __m256i vksum0 = _mm256_mullo_epi32(vacc0, vzeropoint);
+      __m256i vpack0 =  _mm256_loadu_si256((const __m256i*) (packed_b + 0));
+      vpack0 = _mm256_sub_epi32(vpack0, vksum0);
+      _mm256_storeu_si256((__m256i *) (packed_b + 0), vpack0);
+      out = (int8_t*) ((uintptr_t) out + extra_bytes);
+      w0 = w7;
+    }
+
+    // NC remainder (1..7)
+    if XNN_UNLIKELY(n != 0) {
+      assert(n >= 1 && n <= 7);
+
+      int32_t* packed_b = (int32_t*) out;
+      if XNN_LIKELY(b != NULL) {
+        size_t nb = n;
+        do {
+          *((int32_t*) out) = *b++;
+          out += sizeof(int32_t);
+        } while (--nb != 0);
+      } else {
+        size_t nb = n;
+        do {
+          *((int32_t*) out) = 0;
```

Comment:
This can store full vectors of 0 the same as the main NC loop

        _mm256_storeu_si256((__m256i*) (out + 0), _mm256_setzero_si256());

Reply:
that maybe cause memory out of bound? unless add a mask

Reply:
updated. the output is padded, so it will not out of bound. Thanks the feedback